### PR TITLE
Angolasup 661: Allow ticket submissions without user's email address

### PR DESCRIPTION
**Pull request by mjozefow**

### CODE REVIEW

File: src/test/java/org/openlmis/servicedesk/domain/ServiceDeskCustomerDataBuilder.java [hunk 1]
```diff
@@ -37,6 +37,10 @@ public ServiceDeskCustomerDataBuilder() {
     customerId = "customer" + instanceNumber;
   }
 
+  public ServiceDeskCustomerDataBuilder withEmail(String email) {
+    this.email = email;
+    return this;
+  }
   /**
```
Comment (kmadej) [inline src/test/java/org/openlmis/servicedesk/domain/ServiceDeskCustomerDataBuilder.java:45]: add new., empty line

File: src/test/java/org/openlmis/servicedesk/service/IssueServiceTest.java [hunk 2]
```diff
@@ -65,10 +64,12 @@
 import org.openlmis.servicedesk.web.issue.IssueDtoDataBuilder;
 import org.springframework.http.ResponseEntity;
 import org.springframework.mock.web.MockMultipartFile;
+import org.springframework.test.util.ReflectionTestUtils;
 
 @RunWith(MockitoJUnitRunner.class)
 public class IssueServiceTest {
 
+  private static final String userEmail = "user@siglofa.com";
```
Comment (kmadej) [inline src/test/java/org/openlmis/servicedesk/service/IssueServiceTest.java]: static final field names are in uppercase. Moreover private fields should be at the bottom (below public)

File: src/test/java/org/openlmis/servicedesk/service/IssueServiceTest.java [hunk 3]
```diff
@@ -119,6 +120,8 @@ public void setUp() {
         eq(email),
         eq(user.getUsername())))
         .thenReturn(customerRequest);
+
+    ReflectionTestUtils.setField(issueService, "userEmail", userEmail );
```
Comment (kmadej) [inline src/test/java/org/openlmis/servicedesk/service/IssueServiceTest.java]: remove this

File: src/test/java/org/openlmis/servicedesk/service/IssueServiceTest.java [hunk 4]
```diff
@@ -171,13 +174,16 @@ public void shouldThrowExceptionIfCannotCreateAndFindCustomerWithEmail() {
   }
 
   @Test
-  public void shouldThrowExceptionIfUserHasNoEmail() {
-    expectedException.expect(ValidationMessageException.class);
-    expectedException.expectMessage(new Message(CURRENT_USER_HAS_NO_EMAIL).toString());
-
+  public void shouldUseAdminsEmailIfUserHasNoEmail() {
+    ReflectionTestUtils.setField(issueService, "userEmail", userEmail );
```
Comment (kmadej) [inline src/test/java/org/openlmis/servicedesk/service/IssueServiceTest.java]: redundant space after userEmail before )

**Review by kmadej — CHANGES_REQUESTED** [review]

File: src/test/java/org/openlmis/servicedesk/service/IssueServiceTest.java (hunk 2 above)
Comment (kmadej) [reply src/test/java/org/openlmis/servicedesk/service/IssueServiceTest.java]: add change field name. User email indicates that this is the same email than user field, however it's different. So i think it's should be named rather ADMIN_EMAIL

**Review by kmadej — CHANGES_REQUESTED** [review]

File: src/test/java/org/openlmis/servicedesk/domain/ServiceDeskCustomerDataBuilder.java (hunk 1 above)
Comment (mjozefow) [reply src/test/java/org/openlmis/servicedesk/domain/ServiceDeskCustomerDataBuilder.java:45]: Fixed

File: src/test/java/org/openlmis/servicedesk/service/IssueServiceTest.java (hunk 2 above)
Comment (mjozefow) [reply src/test/java/org/openlmis/servicedesk/service/IssueServiceTest.java]: Fixed

File: src/test/java/org/openlmis/servicedesk/service/IssueServiceTest.java (hunk 3 above)
Comment (mjozefow) [reply src/test/java/org/openlmis/servicedesk/service/IssueServiceTest.java]: Fixed

File: src/test/java/org/openlmis/servicedesk/service/IssueServiceTest.java (hunk 4 above)
Comment (mjozefow) [reply src/test/java/org/openlmis/servicedesk/service/IssueServiceTest.java]: Fixed

**Review by kmadej — APPROVED** [review]
looks good to me

**Pull request merged**